### PR TITLE
feat: DialogueRunner for jenny

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

The `DialogueRunner` is the engine that actually runs the dialogue.

Note: documentation will be added in PR #2102.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [-] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: packages/flame_yarn/README.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
 # flame_yarn
 
 Bridge package for [Yarn Spinner](https://docs.yarnspinner.dev).
```
Comment (spydon) [inline packages/flame_yarn/README.md:3]: Maybe add one sentence about what Yarn is?

File: packages/flame_yarn/test/parse/tokenize_test.dart [hunk 2]
```diff
@@ -2,7 +2,7 @@ import 'package:flame_yarn/src/parse/token.dart';
 import 'package:flame_yarn/src/parse/tokenize.dart';
 import 'package:test/test.dart';
 
-import 'utils.dart';
+import '../utils.dart';
```
Comment (spydon) [inline packages/flame_yarn/test/parse/tokenize_test.dart:5]: Does this library not have `flame_lint` yet? :thinking:

**Review by spydon — COMMENTED** [review]
Lgtm! Great dartdocs on the `DialogueRunner`.

**Review by spydon — APPROVED** [review]

File: packages/flame_yarn/test/parse/tokenize_test.dart (hunk 2 above)
Comment (st-pasha) [reply packages/flame_yarn/test/parse/tokenize_test.dart:5]: I think our linter is set up in such a way that it allows relative paths within the test/ directory. See [this test](https://github.com/flame-engine/flame/blob/main/packages/flame/test/rendering/paint_decorator_test.dart) for example.

Comment (spydon) [reply packages/flame_yarn/test/parse/tokenize_test.dart:5]: Aah ofc, you can't even do package imports from tests I think?
And the rule only states "for lib imports".
https://dart-lang.github.io/linter/lints/avoid_relative_lib_imports.html

**Pull request merged**